jctools-build/src/main/java/org/jctools/queues/util/GeneratorUtils.java

```diff
@@ -0,0 +1,94 @@
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.comments.Comment;
+
+import java.io.File;
+import java.io.FileWriter;
+
+public class GeneratorUtils {
+
+    /**
+     * Remove padding fields from the given class.
+     */
+    public static void cleanupPaddingComments(CompilationUnit cu)
+    {
+        for (Comment comment : cu.getAllContainedComments())
+        {
+            String content = comment.getContent();
+            if (content.contains("byte b") || content.contains("drop 8b") || content.contains("drop 16b") ) {
+                comment.remove();
+            }
+        }
+    }
+
+    /**
+     * Remove padding fields from the given class.
+     */
+    public static void removePaddingFields(ClassOrInterfaceDeclaration node) {
+        // remove padding fields
+        for (FieldDeclaration field : node.getFields())
+        {
+            String fieldName = field.getVariables().get(0).getNameAsString();
+            if (fieldName.startsWith("b0") || fieldName.startsWith("b1")) {
+                node.remove(field);
+            }
+        }
+    }
+
+    /**
+     * Build a generator instance of the given class with the given file name.
+     */
+    public static <T> T buildGenerator(Class<? extends T> generatorClass, String fileName) throws Exception {
+        return generatorClass.getDeclaredConstructor(String.class).newInstance(fileName);
+    }
+
+    protected static final String INDENT_LEVEL = "    ";
+
+    /**
+     * Format a multiline javadoc comment with the given lines and indentation level.
+     */
+    public static String formatMultilineJavadoc(int indent, String... lines) {
+        String indentation = "";
+        for (int i = 0; i < indent; i++) {
+            indentation += INDENT_LEVEL;
+        }
+
+        String out = "\n";
+        for (String line : lines) {
+            out += indentation + " * " + line + "\n";
+        }
+        out += indentation + " ";
+        return out;
+    }
+
+    public static void runJCToolsGenerator(Class<? extends JCToolsGenerator> generatorClass, String[] args) throws Exception {
+
+        if (args.length < 2) {
+            throw new IllegalArgumentException("Usage: outputDirectory inputSourceFiles");
+        }
+
+        File outputDirectory = new File(args[0]);
+
+        for (int i = 1; i < args.length; i++) {
+            File file = new File(args[i]);
+            System.out.println("Processing " + file);
+            CompilationUnit cu = new JavaParser().parse(file).getResult().get();
+            JCToolsGenerator generator = buildGenerator(generatorClass, file.getName());
+            generator.visit(cu, null);
+            generator.organiseImports(cu);
+            generator.cleanupComments(cu);
+
+            String outputFileName = generator.translateQueueName(file.getName().replace(".java", "")) + ".java";
```

Comment:
I could make `JCToolsGenerator` to use a single method `String generate(String queueName)` which return the translated queue name instead of having 3 different methods, executing a generic generation process

Reply:
I'm saving to add more abstractions @nitsanw because I don't know yet if this PR is decent enough; IDK if makes sense to create abstractions of any kind, if not for the sake of re-using code, given that generating code isn't a proper domain which require being strongly defined, but I see it more as a serie of transformation steps/scripts which we want to write once and modify after 4-5 years (unless bugs).